### PR TITLE
Add support for auth ticket header

### PR DESCRIPTION
**Pull request by kjetilh33**

### CODE REVIEW

File: src/main/java/com/cognite/client/CogniteClient.java [hunk 1]
```diff
@@ -85,7 +85,8 @@ There is an assumption that all CDF resources (endpoints) have 90 seconds read/w
 
     /**
      * Returns a {@link CogniteClient} using the provided supplier (function) to provide
-     * a bearer token for authorization.
+     * a bearer token for authorization. The token will be placed into the request header:
+     * {@code Authorization: <theTicket>}
```
Comment (cyriaquebrousse) [inline src/main/java/com/cognite/client/CogniteClient.java:89]:
```suggestion
     * {@code auth-ticket: <theTicket>}
```

Comment (kjetilh33) [reply src/main/java/com/cognite/client/CogniteClient.java]: Unclear code docks on token vs ticket. Have made an update

**Review by ngerstle-cognite — COMMENTED** [review]
The cdf-sdk-java library is intended for external use?
Auth tickets are purely for cdf  internal services, so perhaps something less specific?

**Comment by cyriaquebrousse** [comment]
> The cdf-sdk-java library is intended for external use? Auth tickets are purely for cdf internal services, so perhaps something less specific?

Perhaps a way to set headers in general?

**Comment by kjetilh33** [comment]
> > The cdf-sdk-java library is intended for external use? Auth tickets are purely for cdf internal services, so perhaps something less specific?
> 
> Perhaps a way to set headers in general?

Good idea. I'll refactor to a more generic version which will allow for the auth-ticket as well...

**Comment by kjetilh33** [comment]
@cyriaquebrousse , @ngerstle-cognite , I have refactored the client builder to be more generic. It takes a header key + a token supplier (to set the header value per request) as inputs. Better than before?

**Comment by ngerstle-cognite** [comment]
> > The cdf-sdk-java library is intended for external use? Auth tickets are purely for cdf internal services, so perhaps something less specific?
> 
> Perhaps a way to set headers in general?

that works for me, thanks! :smile:

**Review by cyriaquebrousse — APPROVED** [review]
That's great 👌🏻

**Pull request merged**